src/audio/dai.c

```diff
@@ -741,12 +746,18 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 			dd->chan = NULL;
 		}
 		break;
+#if CONFIG_CAVS_ALH
```

Comment:
Is there a reason why ALH is conditionally handled, but not SSP, HDA, ESAI, SAI??
This doesn't seem consistent, either each DAI case is conditional or none are

Reply:
alh.h is only implemented in CNL, ICL and TGL platforms. Without this check FW build will failed on APL and BYT.  Maybe using alh.h is not a good solution.

Reply:
@RanderWang @plbossart we actually need to move the DAI specific code out of generic DAI.c.

Reply:
Yes, I don't see why the Intel code would deal with NXP's SAI and ESAI and vice-versa. Something's not right with the code partition here.

Reply:
created #2269 @RanderWang assigned you initially.